torch/_dynamo/polyfills/tensor.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any
+
+import torch
+
+from ..decorators import substitute_in_graph
+
+
+@substitute_in_graph(  # type: ignore[arg-type]
+    torch.Tensor._make_subclass
+)
+def make_subclass(
+    cls: type[Any], data: torch.Tensor, requires_grad: bool = False, **kwargs: Any
+) -> Any:
+    # This is a rough approximation of `THPVariable_make_subclass`. It should
+    # suffice for most of Dynamo tracing purposes.
+    assert len(kwargs) == 0, "_make_subclass only supports requires_grad as keyword arg"
+    data = data.detach()
+
+    # Avoid unnecessary `requires_grad` mutation, which isn't supported in Dynamo.
+    if data.requires_grad != requires_grad:
+        data.requires_grad = requires_grad
+
```

Comment:
Is it possible to graph break on things we don't support? like the rest of this https://github.com/pytorch/pytorch/blob/4271ebdbdc43143215896cd128ca6eb5c1aea17a/torch/csrc/autograd/python_variable.cpp#L628-L643

(does torch._dynamo.graph_break() in a polyfill work?)

Reply:
Isn't this what the assert above handles?
```python
assert len(kwargs) == 0, "_make_subclass only supports requires_grad as keyword arg"
```

Reply:
What I meant was:
- setting requires_grad=True is probably not OK here. Does the user code require that requires_grad=True is set here? If not, can we graph break if requires_grad=True?
- we are not calling set_python_custom_device, set_python_custom_layout, _change_backend_component_keys in the polyfill. If dispatch_device and dispatch_layout etc are False then this is fine. If the user calls make_subclass with dispatch_device=True then we should probably graph break

Reply:
Ah I see. I think we would graph break (as I intended) in these cases:
> If not, can we graph break if requires_grad=True?

> If dispatch_device and dispatch_layout etc are False then this is fine. If the user calls make_subclass with dispatch_device=True then we should probably graph break

With the assert I pasted above, we are even more strict on the second case, i.e., if user explicitly pass in extra kwarg like `dispatch_device` at all, we'd graph break.

For the first case (setting `requires_grad`), if that happens we'd just hit the following and graph break accordingly.
https://github.com/pytorch/pytorch/blob/1d9343b44400718291cb1e75ee54f3489f64718f/torch/_dynamo/variables/builtin.py#L1928-L1934

Reply:
Fair enough, thanks for walking me through it!